src/p4est_mesh.c

```diff
@@ -1147,6 +1253,23 @@ p4est_mesh_t       *
 p4est_mesh_new_ext (p4est_t * p4est, p4est_ghost_t * ghost,
                     int compute_tree_index, int compute_level_lists,
                     p4est_connect_type_t btype)
+{
+  p4est_mesh_params_t parameters;
+
```

Comment:
If we call p4est_mesh_params_init (&mesh->parameters) we avoid duplicate settings.

Reply:
You're right; I had a slightly different angle. The `mesh->params = *params;` is fine.

I was thinking of using after `/* initialize parameter struct to pass to mesh_new_params */`
something like `p4est_mesh_init_params (&parameters)` and thereafter only overwriting the values that are not default. Would that be safer or do you prefer the current approach?

As a side note, shall we rename any leftover `parameters` to `params` for consistency?

Reply:
That's a good point, initializing first with `p4est_mesh_init_params` ensures that new additions to the `mesh_params_t` do not lead to unexpected behaviour in `mesh_new_ext`. I've changed this and renamed `parameters` to `params` as well.

Reply:
Looks very nice!